runtime/bundles/org.eclipse.core.contenttype/src/org/eclipse/core/internal/content/ContentTypeCatalog.java

```diff
@@ -418,6 +418,15 @@ IContentType[] findContentTypesFor(ContentTypeMatcher matcher, final String file
 		return selected;
 	}
 
+	IContentType findFirstContentTypeFor(ContentTypeMatcher matcher, final String fileName) {
```

Comment:
```suggestion
	Optional<IContentType> findFirstContentTypeFor(ContentTypeMatcher matcher, final String fileName) {
```

this avoids to return `null` values